lib/interpreters/split_clustering.rb

```diff
@@ -614,21 +613,24 @@ def self.update_matrix_index(vrp)
 
     def self.build_partial_service_vrp(service_vrp, partial_service_ids, available_vehicles_indices = nil, entity = nil)
       log '---> build_partial_service_vrp', level: :debug
+      Models.delete_all
       tic = Time.now
       # WARNING: Below we do marshal dump load but we continue using original objects
       # That is, if these objects are modified in sub_vrp then they will be modified in vrp too.
       # However, since original objects are coming from the data and we shouldn't be modifiying them, this doesn't pose a problem.
       # TOD0: Here we do Marshal.load/dump but we continue to use the original objects (and there is no bugs related to that)
       # That means we don't need hard copy of obejcts we just need to cut the connection between arrays (like services points etc) that we want to modify.
       vrp = service_vrp[:vrp]
-      sub_vrp = Marshal.load(Marshal.dump(vrp))
-      sub_vrp.id = Random.new
+      vrp_hash = JSON.parse(vrp.empty_json, symbolize_names: true)
+      sub_vrp = Models::Vrp.create(vrp_hash)
+      sub_vehicles = []
+      sub_routes = []
       # TODO: Within Periodic Vehicles require to have unduplicated ids
       if available_vehicles_indices
-        sub_vrp.vehicles.delete_if.with_index{ |_v, v_i| !available_vehicles_indices.include?(v_i) }
-        sub_vrp.routes.delete_if{ |r|
+        sub_vrp.vehicles = vrp.vehicles.select.with_index{ |_v, v_i| available_vehicles_indices.include?(v_i) }
+        sub_vrp.routes = vrp.routes.select{ |r|
```

Comment:
I am not sure about this. The above warning looks old, and I feel like the behavior change might dangerous, lets discuss this one.

Reply:
I have removed the above comment. The behaviour has change in order to avoid to keep any link between old and duplicated objects

Reply:
but here we keep a link don't we? `sub_vrp.vehicles = vrp.vehicles.select` would use the vehicles of the original vrp in the sub problems, no?

Reply:
yes, we select parts of the upper level vrp, but only to transform these into a json and then a hash using `sub_vrp_hash = JSON.parse(sub_vrp.to_json, symbolize_names: true)`